### PR TITLE
(maint) Remove razor-server.env

### PR DESCRIPTION
**Pull request by mwaggett**
This commit removes references to the razor-server.env file, as it has been
consolidated with razor-server.sysconfig.

### CODE REVIEW

**Comment by mwaggett** [comment]
should go in with https://github.com/puppetlabs/puppetlabs-pe_razor/pull/61 and https://github.com/puppetlabs/razor-server/pull/410

File: configs/components/razor-server.rb [hunk 1]
```diff
@@ -75,16 +74,11 @@
   if settings[:pe_package]
     case platform.servicetype
     when "systemd"
-      # Add JAVA to razor-server.env so it can find pe-java correctly
-      install_commands.push("sed -i '/^LANG=en_US.UTF-8$$/ a JAVA=#{settings[:server_bindir]}/java' #{settings[:prefix]}/razor-server.env")
```
Comment (mwaggett) [inline configs/components/razor-server.rb]: I added JAVA to the sysconfig template in the puppetlabs-pe_razor module. Is that enough? Or should the package have that as well?

Comment (smcclellan) [reply configs/components/razor-server.rb]: I think the package should have it too, unless the package (on upgrades) will overwrite what the module lays down?

File: configs/components/razor-server.json [hunk 2]
```diff
@@ -1 +1 @@
-{"url": "git://github.com/puppetlabs/razor-server.git", "ref": "1.8.1"}
+{"url": "git://github.com/mwaggett/razor-server.git", "ref": "3b8c0d4836330cdaf05656eee368e3db9a8ca2b6"}
```
Comment (smcclellan) [inline configs/components/razor-server.json]: This PR is actually blocked on the razor-server PR, right?

Comment (mwaggett) [reply configs/components/razor-server.json]: yeah, I just added this for testing, will remove when I'm done.

**Comment by mwaggett** [comment]
Tested foss installs on el-7-x86_64 and ubuntu-14.04-amd64.
Test PE upgrades from PE 2018.1.0 on el-6-x86_64 and el-7-x86_64.

**Comment by mwaggett** [comment]
I'll bump the razor-server version after https://github.com/puppetlabs/razor-server/pull/411 goes in.

**Pull request merged**